### PR TITLE
Update base

### PR DESCRIPTION
**Pull request by redxaxder**

### CODE REVIEW

**Review by JaredCorduan — APPROVED** [review]
LGTM

File: semantics/executable-spec/src/Data/AbstractSize.hs [hunk 1]
```diff
@@ -208,7 +218,7 @@ instance HasTypeReps (SignedDSIGN MockDSIGN a) where
   -- and a 'Word64'. For the 'ByteString' representation we return one character
   -- per byte.
   typeReps (SignedDSIGN (SigMockDSIGN (UnsafeHash bs) i)) =
-    typeOf i <| Seq.replicate (BS.length bs) (typeOf (undefined :: Char))
+    typeOf i <| Seq.replicate (SBS.length bs) (typeOf (undefined :: Char))
```
Comment (dcoutts) [inline semantics/executable-spec/src/Data/AbstractSize.hs]: Or don't use the underlying constructor and use `hashToBytes`. Then there's no need to import SBS.

Or use the function that tells you the hash size given just a proxy for the type, without any example value needed.

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs [hunk 2]
```diff
@@ -303,10 +304,12 @@ getRewardAcnt = do
...
 
 getHash :: forall h a. Hash.HashAlgorithm h => Get (Hash.Hash h a)
-getHash = Hash.UnsafeHash <$> B.getByteString (fromIntegral $ Hash.sizeHash ([] @h))
+getHash =
+  Hash.UnsafeHash . SBS.toShort
+    <$> (B.getByteString . fromIntegral $ Hash.sizeHash ([] @h))
 
 putHash :: Hash.Hash h a -> Put
-putHash (Hash.UnsafeHash b) = B.putByteString b
+putHash (Hash.UnsafeHash b) = B.putByteString (SBS.fromShort b)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs]: Or avoid using the constructor and use the conversion functions provided: `hashToBytes` and `hashFromBytes`

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs [hunk 3]
```diff
@@ -427,4 +430,4 @@ bootstrapKeyHash (BootstrapAddress byronAddress) =
   -- from Hash.Blake2b_224)
   let root = Byron.addrRoot byronAddress
       bytes = Byron.abstractHashToBytes root
-   in KeyHash (Hash.UnsafeHash bytes)
+   in KeyHash (Hash.UnsafeHash . SBS.toShort $ bytes)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address.hs]: Use the conversion functions, or I'll deprecate and hide the constructor! ;-)

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address/Bootstrap.hs [hunk 4]
```diff
@@ -165,7 +166,7 @@ bootstrapWitKeyHash ::
   BootstrapWitness crypto ->
   KeyHash 'Witness crypto
 bootstrapWitKeyHash (BootstrapWitness (VKey key) _ (ChainCode cc) (KeyPadding prefix suffix)) =
-  KeyHash . Hash.UnsafeHash . hash_crypto . hash_SHA3_256 $ bytes
+  KeyHash . Hash.UnsafeHash . SBS.toShort . hash_crypto . hash_SHA3_256 $ bytes
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address/Bootstrap.hs]: and here

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Orphans.hs [hunk 5]
```diff
@@ -59,3 +61,6 @@ instance NFData BlockNo
 instance NoUnexpectedThunks WC.XSignature where
   whnfNoUnexpectedThunks ctxt s = whnfNoUnexpectedThunks ctxt (WC.unXSignature s)
   showTypeOf _proxy = "XSignature"
+
+instance SignableRepresentation (Hash.Hash a b) where
+  getSignableRepresentation = Hash.hashToBytes
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Orphans.hs:66]: :tada:

File: shelley/chain-and-ledger/executable-spec/test/Test/Cardano/Crypto/VRF/Fake.hs [hunk 6]
```diff
@@ -138,7 +138,7 @@ evalVRF' a sk@(SignKeyFakeVRF n) =
   let y = sneakilyExtractResult a sk
       p = unsneakilyExtractPayload a
       realValue =
-        fromIntegral . fromHash
+        fromIntegral . bytesToNatural . hashToBytes
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Cardano/Crypto/VRF/Fake.hs:141]: :+1:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs [hunk 7]
```diff
@@ -119,12 +120,12 @@ goldenTests_MockCrypto =
...
-      KeyHashObj . KeyHash . UnsafeHash . fst $
-        B16.decode "01020304"
+      KeyHashObj . KeyHash . UnsafeHash $
+        "01020304"
     scriptHash :: Credential kh (ConcreteCrypto ShortHash)
     scriptHash =
-      ScriptHashObj . ScriptHash . UnsafeHash . fst $
-        B16.decode "05060708"
+      ScriptHashObj . ScriptHash . UnsafeHash $
+        "05060708"
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs]: I think this does not mean what you think it means. You think this means a 4-byte hash with hex 0x01020304, but you're using `UnsafeHash ("01020304" :: ShortByteString)` so you're getting an 8-byte hash with the given ascii characters.

You're trying to use the `IsString` instance, but for that you need to use a string literal at the Hash type, like
```
ScriptHashObj (KeyHash "01020304")
```
So again the moral of the story is don't use the unsafe constructor :grin:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs [hunk 8]
```diff
@@ -191,7 +192,7 @@ goldenTests_ShelleyCrypto =
     -- and should be 28-byte in the aftermath
     keyBlake2b224 :: BS.ByteString -> Credential kh ShelleyCrypto
     keyBlake2b224 vk =
-      KeyHashObj . KeyHash . UnsafeHash $ hk
+      KeyHashObj . KeyHash . UnsafeHash $ SBS.toShort hk
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address.hs]: :raised_eyebrow:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address/Bootstrap.hs [hunk 9]
```diff
@@ -225,7 +226,7 @@ aliceBadWitness =
 bobAddr :: Addr C
 bobAddr = Addr Testnet (KeyHashObj $ coerce someHash) StakeRefNull
   where
-    someHash = "someHash" :: ByteString
+    someHash = "someHash" :: ShortByteString
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Address/Bootstrap.hs]: Why coerce when you can either use the appropriate conversion functions, or the string literal syntax.
```
bobAddr = Addr Testnet (KeyHashObj "736f6d6548617368") StakeRefNull 
```
```
> Base16.encode "someHash"
"736f6d6548617368"
```

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Examples.hs [hunk 10]
```diff
@@ -523,10 +523,12 @@ alicePoolParams =
 --  For our purposes in this test we can bootstrap the chain by just coercing the value.
 --  When this transition actually occurs, the consensus layer will do the work of making
 --  sure that the hash gets translated across the fork
-lastByronHeaderHash :: forall proxy h. HashAlgorithm h => proxy h -> HashHeader h
-lastByronHeaderHash _ = HashHeader $ coerce (hash 0 :: Hash (ConcreteCrypto h) Int)
+lastByronHeaderHash :: forall proxy h. proxy h -> HashHeader h
+lastByronHeaderHash _ =
+  HashHeader $
+    coerce ("0" :: ShortByteString)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Examples.hs]: more unnecessary coercing :raised_eyebrow:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Generator/Genesis.hs [hunk 11]
```diff
@@ -153,7 +154,7 @@ genCredential =
     ]
 
 genHash :: forall c a. HashAlgorithm (HASH c) => Gen (Hash c a)
-genHash = UnsafeHash . BS.pack <$> genWords numBytes
+genHash = UnsafeHash . SBS.toShort . BS.pack <$> genWords numBytes
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Generator/Genesis.hs]: :raised_eyebrow:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Generator/Trace/Chain.hs [hunk 12]
```diff
@@ -80,8 +81,8 @@ instance HashAlgorithm h => HasTrace (CHAIN h) (GenEnv h) where
 -- For our purposes we can bootstrap the chain by just coercing the value.
 -- When this transition actually occurs, the consensus layer will do the work of making
 -- sure that the hash gets translated across the fork
-lastByronHeaderHash :: forall proxy h. HashAlgorithm h => proxy h -> HashHeader h
-lastByronHeaderHash _ = HashHeader $ coerce (hash 0 :: Hash (ConcreteCrypto h) Int)
+lastByronHeaderHash :: forall proxy h. proxy h -> HashHeader h
+lastByronHeaderHash _ = HashHeader $ coerce ("0" :: ShortByteString)
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Generator/Trace/Chain.hs]: I need some variety in my emoji. I'll branch out.

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Orphans.hs [hunk 13]
```diff
@@ -1,15 +1,26 @@
...
+import Shelley.Spec.Ledger.OCert (KESPeriod (..))
 
 -- We need this here for the tests, but should not be in the actual library because
 -- a Num instance for this type does not make sense in the general case.
 deriving instance Num (DSIGN.VerKeyDSIGN (DSIGN crypto)) => Num (VKey kd crypto)
+
+instance (KES.KESAlgorithm c) => SignableRepresentation (KES.VerKeyKES c, Natural, KESPeriod) where
+  getSignableRepresentation (vk, nat, KESPeriod p) =
+    serializeEncoding' $
+      KES.encodeVerKeyKES vk <> toCBOR nat <> toCBOR p
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Orphans.hs:26]: :orange_heart:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Rewards.hs [hunk 14]
```diff
@@ -111,13 +114,19 @@ keyPair :: Crypto c => Int -> KeyPair r c
 keyPair seed = KeyPair vk sk
   where
     vk = VKey (Crypto.deriveVerKeyDSIGN sk)
-    sk = Crypto.genKeyDSIGN $ mkSeedFromBytes . coerce $ hash @MD5 seed
+    sk =
+      Crypto.genKeyDSIGN $
+        mkSeedFromBytes . SBS.fromShort . coerce $
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Rewards.hs]: :see_no_evil:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Rewards.hs [hunk 15]
```diff
@@ -111,13 +114,19 @@ keyPair :: Crypto c => Int -> KeyPair r c
...
+          hashWithSerialiser @MD5 toCBOR seed
 
 vrfKeyPair :: forall v. Crypto.VRFAlgorithm v => Int -> (Crypto.SignKeyVRF v, Crypto.VerKeyVRF v)
 vrfKeyPair seed = (sk, vk)
   where
     vk = Crypto.deriveVerKeyVRF sk
-    sk = Crypto.genKeyVRF $ mkSeedFromBytes . coerce $ hash @MD5 seed
+    sk =
+      Crypto.genKeyVRF $
+        mkSeedFromBytes . SBS.fromShort . coerce $
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Rewards.hs]: :crying_cat_face:

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Utils.hs [hunk 16]
```diff
@@ -98,7 +110,7 @@ mkSeedFromWords ::
   (Word64, Word64, Word64, Word64, Word64) ->
   Seed
 mkSeedFromWords stuff =
-  mkSeedFromBytes . coerce $ hash @MD5 stuff
+  mkSeedFromBytes . SBS.fromShort . coerce $ hashWithSerialiser @MD5 toCBOR stuff
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Utils.hs]: :evergreen_tree:

**Review by dcoutts — CHANGES_REQUESTED** [review]
Maybe I actually should deprecate the constructor.

File: shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Utils.hs (hunk 16 above)
Comment (JaredCorduan) [reply shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Utils.hs]: Duncan, I see you've branched out

File: shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address/Bootstrap.hs [hunk 17]
```diff
@@ -174,8 +175,8 @@ bootstrapWitKeyHash (BootstrapWitness (VKey key) _ (ChainCode cc) (KeyPadding pr
     bytes = prefix <> keyBytes <> cc <> suffix
     hash_SHA3_256 :: ByteString -> ByteString
     hash_SHA3_256 = Hash.digest (Proxy :: Proxy Hash.SHA3_256)
-    hash_crypto :: ByteString -> ByteString
-    hash_crypto = Hash.digest (Proxy :: Proxy (ADDRHASH crypto))
+    hash_crypto :: ByteString -> Hash.Hash (ADDRHASH crypto) a
+    hash_crypto = coerce . Hash.hashWith @(ADDRHASH crypto) id
```
Comment (dcoutts) [inline shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Address/Bootstrap.hs]:
```suggestion
    hash_crypto = Hash.castHash . Hash.hashWith @(ADDRHASH crypto) id
```

**Review by dcoutts — APPROVED** [review]
LGTM!

**Pull request merged**